js/src/forum/components/Search.js

```diff
@@ -19,6 +20,8 @@ import UsersSearchSource from './UsersSearchSource';
  */
 export default class Search extends Component {
   init() {
+    this.state = this.props.state || new SearchState();
```

Comment:
Is the fallback just defensive coding or is this going to happen in known code? If not, let's just require it? (Should also be added to the component's docblock, I guess.)

Reply:
Doesn't happen anywhere I know of; and agree, explicit is better than implicit. As to docblock, yes on adding that, but that'll be Typescript-ized soon enough, hopefully.